algorithms/src/std_algorithms/impl/Kokkos_InclusiveScan.hpp

```diff
@@ -24,6 +24,47 @@
 #include <std_algorithms/Kokkos_Distance.hpp>
 #include <string>
 
+#if defined(KOKKOS_ENABLE_CUDA)
+
+// Workaround for `Instruction 'shfl' without '.sync' is not supported on
+// .target sm_70 and higher from PTX ISA version 6.4`.
+// Also see https://github.com/NVIDIA/cub/pull/170.
+#if !defined(CUB_USE_COOPERATIVE_GROUPS)
+#define CUB_USE_COOPERATIVE_GROUPS
+#endif
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wshadow"
+#pragma GCC diagnostic ignored "-Wsuggest-override"
+
+#if defined(KOKKOS_COMPILER_CLANG)
+// Some versions of Clang fail to compile Thrust, failing with errors like
+// this:
+//    <snip>/thrust/system/cuda/detail/core/agent_launcher.h:557:11:
+//    error: use of undeclared identifier 'va_printf'
+// The exact combination of versions for Clang and Thrust (or CUDA) for this
+// failure was not investigated, however even very recent version combination
+// (Clang 10.0.0 and Cuda 10.0) demonstrated failure.
+//
+// Defining _CubLog here locally allows us to avoid that code path, however
+// disabling some debugging diagnostics
+#pragma push_macro("_CubLog")
+#ifdef _CubLog
+#undef _CubLog
+#endif
+#define _CubLog
+#include <thrust/distance.h>
+#include <thrust/scan.h>
+#pragma pop_macro("_CubLog")
+#else
+#include <thrust/distance.h>
+#include <thrust/scan.h>
+#endif
+
+#pragma GCC diagnostic pop
```

Comment:
Yes, it will be good to run tests on AMD GPUs. I will come back on this.

This code snippet was copied from the sort implementation. Yes, refactoring this part into a separate header is the better approach. Also, as @cedricchevalier19 pointed out, I will check whether the problem still exists with recent Clang and Cuda.